src/main/java/sg/ncl/MainController.java

```diff
@@ -3775,4 +3790,41 @@ private String getUsageStatisticsByTeamId(String id) {
         }
         return response.getBody().toString();
     }
+
+    /**
+     * Invokes the get nodes status in the telemetry service
+     * @return a map containing a list of nodes status by their type
+     */
+    private Map<MachineType, List> getNodesStatus() throws IOException {
+        log.info("Getting all nodes' status from: {}", properties.getNodesStatus());
+
+        Map<MachineType, List> output = new HashMap<>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Convert this Map to an EnumMap. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1640)